doc/releases/changelog-dev.md

```diff
@@ -103,6 +103,9 @@
 
 <h3>Internal changes ⚙️</h3>
 
+* Upgraded source code to use Python 3.11+ language features.
```

Comment:
```suggestion
* Refactored the codebase to adopt modern type hint syntax for Python 3.11+ language features.
```
After all this is a 2k+ changes PR, let's slightly add some highlevel details to the log